### PR TITLE
RHODS-6401 expose grpc port

### PR DESCRIPTION
**Pull request by heyselbi**
## Description

Since routes can be created only with single ports (aka one port per route), creating a separate route for each port. 

## How Has This Been Tested?

It is unfinished, and will be tested once complete.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: controllers/inferenceservice_route.go [hunk 1]
```diff
@@ -93,14 +112,85 @@ func CompareInferenceServiceRoutes(r1 routev1.Route, r2 routev1.Route) bool {
...
+			// log.Error(err, "Unable to add OwnerReference to the Route")
+
+			return false, err
+		}
+
+		// Create the route in the Openshift cluster
+		err = reconciler.Create(ctx, desiredRoute)
+		if err != nil && !apierrs.IsAlreadyExists(err) {
+			// FIXME
+			// log.Error(err, "Unable to create the Route")
```
Comment (Xaenalt) [inline controllers/inferenceservice_route.go:142]: I'd reenable these, just to print a diagnostic message if something goes wrong

File: controllers/inferenceservice_route.go [hunk 2]
```diff
@@ -112,18 +202,27 @@ func (r *OpenshiftInferenceServiceReconciler) reconcileRoute(inferenceservice *i
...
 	// Generate the desired route
-	desiredRoute := newRoute(inferenceservice, enableAuth)
+	if enableAuth {
+		desiredRoute := newRoute(inferenceservice, RouteOptions.ENABLE_AUTH)
+	} else {
+		desiredRoute := newRoute(inferenceservice, RouteOptions.DEFAULT)
+	}
 
-	// Create the route if it does not already exist
+	// Create the default route (possible with auth enable) if it does not already exist
```
Comment (Xaenalt) [inline controllers/inferenceservice_route.go:225]: Nit: possibly

File: controllers/inferenceservice_route.go [hunk 3]
```diff
@@ -65,18 +73,29 @@ func NewInferenceServiceRoute(inferenceservice *inferenceservicev1.InferenceServ
...
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+			}
+		} else if routeOptions.ENABLE_GRPC {
+			finalRoute.Spec.Port = &routev1.RoutePort{
+				TargetPort: intstr.FromInt(modelmeshGrpcPort),
+			}
+			// INFO: Does gRPC need to enable TLS? 
+			finalRoute.Spec.TLS = &routev1.TLSConfig{
+				Termination:                   routev1.TLSTerminationReencrypt,
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
```
Comment (Jooho) [inline controllers/inferenceservice_route.go:97]: GRPC does not need path `"/v2/models/" + inferenceservice.Name`

**Comment by heyselbi** [comment]
closing the request as it turns out gRPC port does not have to be exposed via route and access to it via service internally is sufficient.

**Pull request closed without merging**